### PR TITLE
Expand Adventure Map Event structure for the future support

### PR DESCRIPTION
**Pull request by ihhub**
relates to #6845

Also, move rumors before metadata to have a neat structure.

This is a map breaking change.

### CODE REVIEW

File: src/fheroes2/maps/map_format_info.h [hunk 1]
```diff
@@ -192,6 +192,19 @@ namespace Maps::Map_Format
...
         Funds resources;
+
+        int16_t attack{ 0 };
+        int16_t defense{ 0 };
+        int16_t knowledge{ 0 };
+        int16_t spellPower{ 0 };
+
+        int32_t experience{ 0 };
+
+        int32_t secondarySkill{ 0 };
```
Comment (Districh-ru) [inline src/fheroes2/maps/map_format_info.h:203]: Hi, @ihhub This variable can be int8_t.
The same is for 117 line.

File: src/fheroes2/maps/map_format_info.h [hunk 2]
```diff
@@ -192,6 +192,19 @@ namespace Maps::Map_Format
...
+        uint8_t secondarySkillLevel{ 0 };
+
+        std::array<int32_t, 5> monsterType{ 0 };
+        std::array<int32_t, 5> monsterCount{ 0 };
```
Comment (idshibanov) [inline src/fheroes2/maps/map_format_info.h:207]: Is this for combat or additional reward type?

Comment (ihhub) [reply src/fheroes2/maps/map_format_info.h:207]: Reward type. I think we can allow to give only one type of monsters for simplification.

File: src/fheroes2/maps/map_format_info.h (hunk 1 above)
Comment (ihhub) [reply src/fheroes2/maps/map_format_info.h:203]: Sure!

**Review by idshibanov — APPROVED** [review]

**Review by Districh-ru — APPROVED** [review]

**Pull request merged**